### PR TITLE
feat/flit support

### PR DESCRIPTION
**Pull request by blast-hardcheese**
Why
===

As we continue to expand into the realm of offering better Python package manager support, we're happy to add `flit` to upm.

What changed
============

- Bump `BurntSushi/toml` to 1.4.0
- Broke out `python/poetry.go` and `python/pip.go` to go along with `python/flit.go`
- Fix a bug with pip `requirements.txt` additions where we would take a zero-byte `requirements.txt` and turn it into `\ndep` instead of `dep\n`
- Adding `QuirksCannotAddRemove` to support `flit`, which does not directly support `add` or `remove`. In the future, we can or delete from `pyproject.toml`

Test plan
=========

- Import `https://github.com/mwaskom/seaborn` into Replit
- Run `pip install flit` to get a compatible version of flit
- Run `upm install`
- Verify that dependencies are installed

### CODE REVIEW

File: internal/backends/python/python.go
Comment (cdmistman) [inline internal/backends/python/python.go:1]: 🐐

File: internal/cli/cmds.go [hunk 1]
```diff
@@ -267,6 +267,11 @@ func runAdd(
 	defer span.Finish()
 	b := backends.GetBackend(ctx, language)
 
+	if !b.QuirksCanAddRemove() {
+		fmt.Println("This backend does not suppport dependency additions")
```
Comment (cdmistman) [inline internal/cli/cmds.go:271]:
```suggestion
		fmt.Println("This backend does not suppport dependency addition")
```
nit to keep consistent with the other error message

**Review by cdmistman — CHANGES_REQUESTED** [review]
integration tests please! lgtm so far though

**Comment by blast-hardcheese** [comment]
A workaround was found for flit. This work can be brought back if necessary.

**Pull request closed without merging**